packages/worker/src/api/routes/global/auth.js

```diff
@@ -0,0 +1,61 @@
+const Router = require("@koa/router")
+const authController = require("../../controllers/global/auth")
+const joiValidator = require("../../../middleware/joi-validator")
+const Joi = require("joi")
+
+const router = Router()
+
+function buildAuthValidation() {
+  // prettier-ignore
+  return joiValidator.body(Joi.object({
+    username: Joi.string().required(),
+    password: Joi.string().required(),
+  }).required().unknown(false))
+}
+
+function buildResetValidation() {
+  // prettier-ignore
+  return joiValidator.body(Joi.object({
+    email: Joi.string().required(),
+  }).required().unknown(false))
+}
+
+function buildResetUpdateValidation() {
+  // prettier-ignore
+  return joiValidator.body(Joi.object({
+    resetCode: Joi.string().required(),
+    password: Joi.string().required(),
+  }).required().unknown(false))
+}
+
+router
+  .post(
+    "/api/global/auth/:tenantId/login",
+    buildAuthValidation(),
+    authController.authenticate
+  )
+  .post(
+    "/api/global/auth/:tenantId/reset",
+    buildResetValidation(),
+    authController.reset
+  )
+  .post(
+    "/api/global/auth/:tenantId/reset/update",
+    buildResetUpdateValidation(),
+    authController.resetUpdate
+  )
+  .post("/api/global/auth/logout", authController.logout)
+  .get("/api/global/auth/:tenantId/google", authController.googlePreAuth)
+  .get("/api/global/auth/:tenantId/google/callback", authController.googleAuth)
+  .get(
+    "/api/global/auth/:tenantId/oidc/configs/:configId",
+    authController.oidcPreAuth
+  )
+  .get("/api/global/auth/:tenantId/oidc/callback", authController.oidcAuth)
+  // deprecated - used by the default system before tenancy
+  .get("/api/global/auth/google", authController.googlePreAuth)
+  .get("/api/global/auth/google/callback", authController.googleAuth)
+  .get("/api/global/auth/oidc/configs/:configId", authController.oidcPreAuth)
+  .get("/api/global/auth/oidc/callback", authController.oidcAuth)
```

Comment:
I think we could get away with the two callback urls only here. 
Should we keep them as `/api/admin/` as before for backwards compatibility?

Reply:
ah yes, I forgot about the admin backwards compat too, will add it in!